fastavro/_write.pyx

```diff
@@ -168,6 +172,31 @@ cdef write_array(bytearray fo, list datum, schema, dict named_schemas, fname):
     write_long(fo, 0)
 
 
+cdef write_python_array(bytearray fo,
+                        array.array datum,
+                        schema,
+                        named_schemas,
+                        fname):
+    """Array specialization for python arrays."""
+    if len(datum) > 0:
+        write_long(fo, len(datum))
+        record_type = extract_record_type(schema['items'])
+        if record_type in ('int', 'long'):
+            for idx in range(len(datum)):
```

Comment:
Can these loops be changed to the following?

```python
for item in datum:
    write_X(fo, item)
```

Or does that change the performance?

Reply:
The line-score for
```for entry in datum```
reduces from 43 to 5, but the line-score for
```write_int(fo, entry)```
increases from 1 to 4, so there's probably a tipping-pont somewhere for larger arrays.
When I test it roughly, the difference is below 2%, with the advantage tipping somewhere between 50 and 200 array-size, so I'll go with the easiest to read.